pkgs/servers/peertube/default.nix

```diff
@@ -2,14 +2,21 @@
 , brotli, fixup_yarn_lock, jq, nodejs, which, yarn
 }:
 let
-  arch =
-    if stdenv.hostPlatform.system == "x86_64-linux" then "linux-x64"
-    else throw "Unsupported architecture: ${stdenv.hostPlatform.system}";
-
+  bcrypt_arch_details = {
+    "x86_64-linux" = {
+      "arch" = "linux-x64";
+      "hash" = "sha256-I1ceMi7h6flvKBmMIU1qjAU1S6z5MzguHDul3g1zMKw=";
+    };
+    "aarch64-linux" = {
+      arch = "linux-arm64";
+      hash = "sha256-q8BR7kILYV8i8ozDkpcuKarf4s1TgRqOrUeLqjdWEQ0=";
+    };
+  };
```

Comment:
Missing or unsupported system

Reply:
I don't understand, do you expect arch = "aarch64-linux"? The string for "arch" is only used as part of the URL for bcrypt_lib below.

Reply:
This code is not evaling on any other platform which it should. see eg https://github.com/NixOS/nixpkgs/blob/68196a61c26748d3e53a6803de3d2f8c69f27831/pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix#L120